model_analyzer/reports/report_manager.py

```diff
@@ -423,6 +426,7 @@ def _find_best_configs(self, report_key):
     def _create_constraint_string(self, report_key: str) -> str:
         constraint_strs = self._build_constraint_strings()
 
+        constraint_str = "None"
```

Comment:
This change was made for fixing the following bug:
```
File "/home/nvex/saikiran/triton_related/initial_rampup/editable_ma/github_branch/model_analyzer/model_analyzer/reports/report_manager.py", line 438, in _create_constraint_string
    return constraint_str
UnboundLocalError: local variable 'constraint_str' referenced before assignment
```
The above error happens when we run profiling if a checkpoint already exists and if two profile models are passed with only one having constraints (no global constraints as well).

Reply:
I'd like to understand this case better. I don't see how we can have a case where global/default constraints do not exist.